doc/source/whatsnew/v0.23.4.txt

```diff
@@ -32,6 +32,7 @@ Bug Fixes
 
 - Bug where calling :func:`DataFrameGroupBy.agg` with a list of functions including ``ohlc`` as the non-initial element would raise a ``ValueError`` (:issue:`21716`)
 - Bug in ``roll_quantile`` caused a memory leak when calling ``.rolling(...).quantile(q)`` with ``q`` in (0,1) (:issue:`21965`)
+- Passing a non-int scalar to ``.iloc`` now raises a more appropriate error message (:issue:`21982`)
```

Comment:
Not sure if it is worth to add a whatsnew for this.

Reply:
OK, removing